src/App.tsx

```diff
@@ -1,61 +1,124 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo, User } from './components/Types/Types';
+
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId)
+    || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [newTodos, setTodos] = useState<Todo[]>(todos);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [userID, setUserID] = useState(0);
+  const [titleError, setTitleError] = useState('');
+  const [userIDError, setUserIDError] = useState('');
+
+  function getId() {
+    const max = Math.max(...newTodos.map(todo => todo.id));
+
+    return max + 1;
+  }
+
+  function resetForm() {
+    setTodoTitle('');
+    setUserID(0);
+    setTitleError('');
+    setUserIDError('');
+  }
+
+  const todoFormHandler = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!userID) {
+      setUserIDError('Username is required');
+    }
+
+    if (!todoTitle) {
+      setTitleError('Title is required');
+    }
+
+    if (!userID || !todoTitle) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: getId(),
+      title: todoTitle,
+      completed: false,
+      userId: userID,
+      user: getUserById(userID),
+    };
+
+    setTodos(prev => [...prev, newTodo]);
+
+    resetForm();
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form onSubmit={e => todoFormHandler(e)}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <input
+            type="text"
+            data-cy="titleInput"
+            id="titleInput"
+            placeholder="Type todo title"
+            value={todoTitle}
+            onChange={e => {
+              setTodoTitle(e.target.value);
+              setTitleError('');
+            }}
```

Comment:
It is necessary for you to follow the checklist.
![image](https://github.com/mate-academy/react_add-todo-form/assets/24822800/d0fed327-47dc-4bab-9a2d-6e90b46635c0)

Replace it wherever you find it